packages/repository/src/model.ts

```diff
@@ -207,7 +207,8 @@ export abstract class Model {
     }
 
     const copyPropertyAsJson = (key: string) => {
-      json[key] = asJSON((this as AnyObject)[key]);
+      const jsonKey = def.properties[key]?.json?.name ?? key;
+      json[jsonKey] = asJSON((this as AnyObject)[key]);
```

Comment:
What about accepting JSON objects from HTTP layer as instances of the model? For example, I would expect the following to be allowed.

```
POST /accounts

{
   account_id: 1
}
```

Reply:
Sounds good. Since I don't use this case how do I go around implementing this? Update the asObject() method for this usecase?

Reply:
Maybe introduce a method `fromJSON` and use it in https://github.com/strongloop/loopback-next/blob/e8c96f97b24cb87e09b1f0336ff2a9dd6292a893/packages/repository/src/model.ts#L250.

Reply:
+1 to introduce `fromJSON` to accompany `toJSON`.

IMO, it's important to honor `json.name` properties both when producing HTTP response bodies but also when parsing incoming HTTP request bodies.